src/core/lib/support/vector.h

```diff
@@ -19,13 +19,65 @@
 #ifndef GRPC_CORE_LIB_SUPPORT_VECTOR_H
 #define GRPC_CORE_LIB_SUPPORT_VECTOR_H
 
-#include "absl/container/inlined_vector.h"
 #include "src/core/lib/support/memory.h"
 
 namespace grpc_core {
 
+// NOTE: We eventually want to use absl::InlinedVector here.  However,
+// there are currently build problems that prevent us from using absl.
+// In the interim, we define a custom implementation as a place-holder,
+// with the intent to eventually replace this with the absl
+// implementation.
+//
+// This place-holder implementation does not implement the full set of
+// functionality from the absl version; it has just the methods that we
+// currently happen to need in gRPC.  If additional functionality is
+// needed before this gets replaced with the absl version, it can be
+// added, with the following proviso:
+//
+// ANY METHOD ADDED HERE MUST COMPLY WITH THE INTERFACE IN THE absl
+// IMPLEMENTATION!
+//
+// TODO(ctiller, nnoble, roth): Replace this with absl::InlinedVector
+// once we integrate absl into the gRPC build system in a usable way.
 template <typename T, size_t N>
-using InlinedVector = absl::InlinedVector<T, N, Allocator<T>>;
+class InlinedVector {
+ public:
+  InlinedVector() {}
+  ~InlinedVector() { gpr_free(dynamic_); }
+
+  // For now, we do not support copying.
+  InlinedVector(const InlinedVector&) = delete;
+  InlinedVector& operator=(const InlinedVector&) = delete;
+
+  T& operator[](size_t offset) {
+    GPR_ASSERT(offset < size_);
+    if (offset < N) {
+      return inline_[offset];
+    } else {
+      return dynamic_[offset - N];
+    }
+  }
+
+  void push_back(const T& value) {
+    if (size_ < N) {
+      inline_[size_] = value;
+    } else {
+      const size_t dynamic_size = size_ - N;
+      dynamic_ = reinterpret_cast<T*>(
```

Comment:
This is not typesafe.

We want to malloc, move construct old values in, and then destroy the old values.

Something like:

```c++
auto new_dynamic = static_cast<T*>(gpr_malloc(...));
for (size_t i = 0; i < dynamic_size; i++) {
  new (&new_dynamic[i]) T(std::move(dynamic_[i]));
  dynamic_[i]->~T();
}
gpr_free(dynamic_);
dynamic_ = new_dynamic;
```